src/shared/auth/ip.guard.ts

```diff
@@ -15,6 +17,19 @@ export class IpGuard implements CanActivate {
   async canActivate(context: ExecutionContext): Promise<boolean> {
     const req = context.switchToHttp().getRequest();
     const ip = getClientIp(req);
+
+    const callLimit = this.getUrlLimit(req.url);
+
+    if (callLimit > 0 && !Config.knownRequestIps.includes(ip)) {
+      const ipWithoutLastElement = ip.split('.').slice(0, -1).join('.');
+      const callCount = await this.ipLogRepo.count({
+        url: req.url,
+        ip: Like(`${ipWithoutLastElement}%`),
+        created: MoreThan(Util.daysBefore(1)),
+      });
+      if (callCount > callLimit) throw new HttpException(`Too many requests ${req.url}`, 429);
+    }
```

Comment:
I think it would be good to have an environment setting to disable rate limiting for some special cases (events, etc).